src/pyFAI/test/test_geometry_refinement.py

```diff
@@ -855,6 +858,24 @@ def test_synthetic(self):
             self.assertAlmostEqual(ref[key][0], r2.__getattribute__(key), delta=ref[key][1],
                                    msg="%s is %s, I expected %s%s%s" % (key, r2.__getattribute__(key), ref[key], os.linesep, r2))
 
+        # test the copy
+        self.assertEqual(r3.calibrant, r2.calibrant)
+        self.assertTrue(numpy.all(r3.data==r2.data))
+        r3.guess_poni()
+        r3.refine2(10000000, fix=[])
+        for key in ref.keys():
+            self.assertAlmostEqual(r3.__getattribute__(key), r2.__getattribute__(key), delta=ref[key][1],
+                                   msg="%s is %s, I expected %s%s%s" % (key, r3.__getattribute__(key), ref[key], os.linesep, r3))
+
+        # test the deep-copy
+        self.assertEqual(r4.calibrant, r2.calibrant)
+        self.assertTrue(numpy.all(r4.data == r2.data))
```

Comment:
One could add a small mutation test? `r4.data[...]= something` and assert `r2.data[...]` remains unchanged.